waveorder/io/reader.py

```diff
@@ -0,0 +1,208 @@
+logging.basicConfig(
+    level=logging.DEBUG,
+    format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+)
+log = logging.getLogger(__name__)
+
+###############################################################################
+
+
+class MicromanagerReader:
+
+    def __init__(self, folder, extract_data=False, log_level=logging.ERROR):
+        """
+        loads ome-tiff files in folder into zarr or numpy arrays
+        Strategy:
+            1. read all files in folder
+            2. search for the file whose omexml does not contain the element tag "BinaryOnly"-> this is the master ome-tiff
+            3. each series represents a micro-manager stage position.  upon call for data, assign this data to a dict
+
+        :param folder: folder containing all ome-tiff files
+        :param reader: tifffile or aicsimageio
+        """
+
+        logging.basicConfig(
+            level=log_level,
+            format="[%(levelname)4s: %(module)s:%(lineno)4s %(asctime)s] %(message)s",
+        )
+        self.log = logging.getLogger(__name__)
+
+        self._positions = {}
+        self.mm_meta = None
+        self.stage_positions = 0
+        self.height = 0
+        self.width = 0
+        self.frames = 0
+        self.slices = 0
+        self.channels = 0
+        self.shape = (self.frames, self.slices, self.channels, self.height, self.width)
+        self.chnames = []
+
+        self.master_ome_tiff = self._get_master_ome_tiff(folder)
+        self._set_mm_meta()
+
+        if extract_data:
+            self.extract_data(self.master_ome_tiff)
+
+    def _set_mm_meta(self):
+        with TiffFile(self.master_ome_tiff) as tif:
+            self.mm_meta = tif.micromanager_metadata
+
+            if self.mm_meta['Summary']['Positions'] > 1:
+                self.stage_positions = []
+                for p in range(self.mm_meta['Summary']['Positions']):
+                    pos = self._simplify_stage_position(self.mm_meta['Summary']['StagePositions'][p])
+                    self.stage_positions.append(pos)
+
+            for ch in self.mm_meta['Summary']['ChNames']:
+                self.chnames.append(ch)
+
+            self.height = self.mm_meta['Summary']['Height']
+            self.width = self.mm_meta['Summary']['Width']
+            self.frames = self.mm_meta['Summary']['Frames']
+            self.slices = self.mm_meta['Summary']['Slices']
+            self.channels = self.mm_meta['Summary']['Channels']
+
+    def _simplify_stage_position(self, stage_pos: dict):
+        """
+        flattens the nested dictionary structure of stage_pos and removes superfluous keys
+        :param stage_pos: dictionary containing a single position's device info
+        :return:
+        """
+        out = copy(stage_pos)
+        out.pop('DevicePositions')
+        for dev_pos in stage_pos['DevicePositions']:
+            out.update({dev_pos['Device']: dev_pos['Position_um']})
+        return out
+
+    def extract_data(self, master_ome):
```

Comment:
Shouldn't this also be an internal function (i.e. `_extract_data`)? Also, I'm not sure about the name of the function. Right now, it sounds like it extract byte data, while in practice it simply creates zarr stopres for every position. Maybe call it `_create_zarr_store`?